sphinx_gallery/gen_rst.py

```diff
@@ -714,7 +714,8 @@ def rst_blocks(script_blocks, output_blocks, file_conf, gallery_conf):
                     example_rst += "\n\n|\n\n"
                 example_rst += code_rst
         else:
-            example_rst += bcontent + '\n'
+            block_separator = '\n\n' if not bcontent.endswith('\n') else '\n'
+            example_rst += bcontent + block_separator
```

Comment:
Can't we always just insert `\n\n`? Too many newlines probably isn't a problem

Reply:
That is true. However, it seems that all blocks except the docstring do actually have a trailing newline. While having more lines is not a problem. I didn't want the PR to introduce more newlines in the majority of cases. If you want to, I can still simplify to always add `\n\n`.

Note: This is the result of the loose definition of the block content. If one wants to do this really clean, trailing newlines on blocks should be consistent (probably be absent).